### PR TITLE
Throw response timeout

### PR DESCRIPTION
**Pull request by lukasz-falda**
AC
---
* reorginize exception throwing
* throw timeout exception

### CODE REVIEW

File: src/LuigisBoxBundle/Service/Client.php [hunk 1]
```diff
@@ -4,36 +4,80 @@
...
+            $retryAfter = reset($retryAfter);
+            throw new TooManyRequestsException((int) $retryAfter, $response);
+        }
+        if (Response::HTTP_REQUEST_ENTITY_TOO_LARGE === $response->getStatusCode()) {
+            throw new TooManyItemsException(null, null, $response);
+        }
+
+        if (Response::HTTP_BAD_REQUEST <= $response->getStatusCode()) {
+            throw RequestException::create($request, $response);
+        }
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/Service/Client.php:69]: move basic exceptions here, and add `RequestException::create` which is missing

**Review by malarzm — APPROVED** [review]

File: src/LuigisBoxBundle/Service/Client.php [hunk 2]
```diff
@@ -4,36 +4,80 @@
...
+        if (Response::HTTP_TOO_MANY_REQUESTS === $response->getStatusCode()) {
+            $retryAfter = $response->getHeader('Retry-After');
+            $retryAfter = reset($retryAfter);
+            throw new TooManyRequestsException((int) $retryAfter, $response);
+        }
+        if (Response::HTTP_REQUEST_ENTITY_TOO_LARGE === $response->getStatusCode()) {
+            throw new TooManyItemsException(null, null, $response);
+        }
+
```
Comment (jrawska) [inline src/LuigisBoxBundle/Service/Client.php]: formatting - why empty line here?

File: tests/Unit/Service/RequestExceptionsTest.php [hunk 3]
```diff
@@ -99,29 +100,39 @@ public function serviceUnavailableThrows(): void
 
     private function getService(Response $expectedResponse): RequestInterface
     {
-        $client = $this->createMock(Client::class);
-        $client->expects($this->once())
-            ->method('request')
+        $guzzleClient = $this->createMock(\GuzzleHttp\Client::class);
```
Comment (jrawska) [inline tests/Unit/Service/RequestExceptionsTest.php:103]: why don't test GuzzleClient like in here: https://github.com/answear/wide-eyes-bundle/blob/master/tests/Unit/Service/AbstractClientTest.php ?

File: src/LuigisBoxBundle/Service/Client.php (hunk 2 above)
Comment (lukasz-falda) [reply src/LuigisBoxBundle/Service/Client.php]: above we have custom exceptions, below guzzle exception. This is kind of separation.
But really - I dont know :)

I will remove it

File: tests/Unit/Service/RequestExceptionsTest.php (hunk 3 above)
Comment (lukasz-falda) [reply tests/Unit/Service/RequestExceptionsTest.php:103]: I'll look at it later, thanks

**Review by jrawska — APPROVED** [review]

**Pull request merged**